module/modmanager/manager.go

```diff
@@ -335,6 +418,19 @@ func (m *module) startProcess(ctx context.Context, parentAddr string, logger gol
 	return nil
 }
 
+func (m *module) stopProcess() error {
+	if m.process == nil {
+		return nil
+	}
+
+	if err := m.process.Stop(); err != nil {
+		return err
+	}
+
+	utils.UncheckedErrorFunc(func() error { return os.Remove(m.addr) })
```

Comment:
It might be better to remove the module's `.sock` file here instead of in `module.go` or the analogous `module.py` and `module/service.cpp`; Python and C++ were both missing this logic. I could add it to both SDKs, but it might make more sense to remove the `.sock` file where we're managing modules. 

With this change, reconfiguration now works correctly in Python (updated my larger testing comment).

Reply:
As noted on the change in module.go, I think removal of the socket should remain the responsibility of the process that created it in the first place. Having a backup check here isn't a bad idea though, but... as it causes logging if/when it was already cleaned up properly, I'd suggest expanding the lambda here to do a quick check. E.g. only remove if the file still exists, so there's one less spurious "error" showing in the logs.

Reply:
That sounds good to me. Expanded the closure here to check if the file exists before removing.